### PR TITLE
Feature/sof 4657 - Add dataset split tests

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: DigitalEcosystem/data_splitting/bg_extra_descriptors.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    warnings.simplefilter('ignore')
+    df['global_instability'] = df.ox_struct.swifter.apply(maybe_global_instability).copy()
+
+
+# In[]:
+
+
+neighbor_finder = JmolNN()
+
+def average_bond_length(structure, featurizer = AverageBondLength(neighbor_finder)):
```
Comment (timurbazhirov) [inline DigitalEcosystem/data_splitting/bg_extra_descriptors.py:123]: We should start building out content like this in a separate file ("features.py"?) and import from there

File: DigitalEcosystem/data_splitting/bg_extra_descriptors.py [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+for allowed_symbols, set_name in [
+    [["N"], "nitrides"],
+    [["O"], "oxides"],
+    [["S"], "sulfides"],
+    [["C"], "carbides"],
+    [["F", "Cl", "Br", "Cl"], "halides"],
+    [["O", "S", "Se", "Te"], "chalcogens"],
+    [["N", "O", "As", "Sb"], "pnictogens"],
+    [["Al", "Ga", "Ge", "In", "Sn", "Sb", "Tl", "Pb", "Bi", "Po"], "pBlockMetals"]
```
Comment (timurbazhirov) [inline DigitalEcosystem/data_splitting/bg_extra_descriptors.py:211]: The naming convention as above can be defined and reused from a separate file too.

File: DigitalEcosystem/data_splitting/bg_extra_descriptors.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+total = len(df)
+print(f"Starting with {total} entries. Includes top-down and bottom-up.")
+      
+# Throw out systems that are predicted to decompose
+df = df[df["decomposition_energy (eV/atom)"] == 0]
+print(f"Discarding {total-len(df)} entries predicted to decompose (Decomp energy != 0). Total is now {len(df)}.")
+total = len(df)
+
+# Remove systems containing f-block / synthetic elements
+fblock_and_synth = ["La", "Ce", "Pr", "Nd", "Pm", "Sm", "Eu", "Gd", "Tb", "Dy", "Ho", "Er", "Tm", "Yb", "Lu",
```
Comment (timurbazhirov) [inline DigitalEcosystem/data_splitting/bg_extra_descriptors.py:52]: This could also be done as a reusable functaion

**Comment by AcylSilane** [comment]
Closing, because this work is going into the "moves" branch of 4659

**Pull request closed without merging**